cognitive/src/main/scala/com/microsoft/azure/synapse/ml/cognitive/SpeakerEmotionInference.scala

```diff
@@ -0,0 +1,133 @@
+import spray.json.DefaultJsonProtocol.StringJsonFormat
+import spray.json._
+
+trait HasLocaleCol extends HasServiceParams {
+  val locale = new ServiceParam[String](this,
+    "locale",
+    s"The locale of the input text",
+    isRequired = true)
+
+  def setLocale(v: String): this.type = setScalarParam(locale, v)
+
+  def setLocaleCol(v: String): this.type = setVectorParam(locale, v)
+}
+
+trait HasVoiceCol extends HasServiceParams {
+  val voice = new ServiceParam[String](this,
+    "voice",
+    s"The name of the voice used for synthesis",
+    isRequired = true)
+
+  def setVoiceName(v: String): this.type = setScalarParam(voice, v)
+
+  def setVoiceNameCol(v: String): this.type = setVectorParam(voice, v)
+}
+
+trait HasInput extends HasServiceParams {
+  val input = new ServiceParam[String](this,
+    "input",
+    s"The text input to annotate with inferred emotion")
+
+  def setInput(v: String): this.type = setScalarParam(input, v)
+
+  def setInputCol(v: String): this.type = setVectorParam(input, v)
+}
+
+trait HasSpeakerEmotionInferenceParams extends HasServiceParams
+  with HasLocaleCol with HasVoiceCol with HasSetLocation
+  with HasInput
+
+object SpeakerEmotionInference
+  extends ComplexParamsReadable[SpeakerEmotionInference] with Serializable
+
+class SpeakerEmotionInference(override val uid: String) extends CognitiveServicesBase(uid)
+  with HasSpeakerEmotionInferenceParams with HasCognitiveServiceInput
+  with HasInternalStringOutputParser
+  with BasicLogging {
+  logClass()
+
+  def this() = this(Identifiable.randomUID(classOf[SpeakerEmotionInference].getSimpleName))
+
+  setDefault(
+    locale -> Left("en-US"),
+    voice -> Left("en-US-SaraNeural"),
+    input -> Left(this.uid + "_input"),
+  )
+
+  def urlPath: String = "cognitiveservices/v1"
+
+  protected val additionalHeaders: Map[String, String] = Map[String, String](
+    ("X-Microsoft-OutputFormat", "textanalytics-json"),
+    ("Content-Type", "application/ssml+xml")
+  )
+
+  override def inputFunc(schema: StructType): Row => Option[HttpRequestBase] = super.inputFunc(schema).andThen(r => {
+    r.map(req => {
+      additionalHeaders.foreach(header => req.setHeader(header._1, header._2))
+      req
+    })
+  })
+
+  override def setLocation(v: String): this.type = {
+    val domain = getLocationDomain(v)
+    setUrl(s"https://$v.tts.speech.microsoft.$domain/cognitiveservices/v1")
+  }
+
+  override protected def prepareEntity: Row => Option[AbstractHttpEntity] = { row =>
+    val body: String =
+      s"<speak version='1.0' xmlns='http://www.w3.org/2001/10/synthesis' xmlns:mstts='https://www.w3.org/2001/mstts'" +
+        s" xml:lang='en-US'><voice name='Microsoft Server Speech Text to Speech Voice (en-US, JennyNeural)'>" +
+        s"<mstts:task name ='RoleStyle'/>${getValue(row, input)}</voice></speak>"
+    Some(new StringEntity(body))
+  }
+
+  def formatSSML(txt: String, lcl: String, vc: String, rsp: String): String = {
```

Comment:
nit: please make these APIs private[ml] if they arent intented for public consumption